tools/mkldscript.c

```diff
@@ -51,11 +51,13 @@ static void write_ld_script(FILE *fout)
                   "    ..%s ", seg->name, seg->name, seg->name, seg->name);
 
         if (seg->fields & (1 << STMT_after))
-            fprintf(fout, "_%sSegmentEnd ", seg->after);
+            fprintf(fout, "(_%sSegmentEnd + %i) & ~ %i ", seg->after, seg->align - 1, seg->align - 1);
```

Comment:
`ALIGN(segmentEnd, align)`
Also only if align is set

Reply:
This didn't seem to work:
`mips-linux-gnu-ld:build/gc-eu-mq-dbg/oot-gc-eu-mq-dbg.ld:3924: non constant or forward reference address expression for section ..code`